### DISCUSSION TITLE
ValueError: 'no permission to update GPIO'

### DISCUSSION THREAD

**Discussion by pitosalas** [gpiozero/gpiozero#1180] [General]
###  Problem:I need to access an ultrasonic distance sensor and would like to use GPioZero. The code is:

```
from gpiozero.pins.pigpio import PiGPIOFactory
from gpiozero import DistanceSensor
from time import sleep

fact = PiGPIOFactory()
sensor = DistanceSensor(echo="GPIO0", trigger="GPIO1", pin_factory=fact)

while True:
    print('Distance: ', sensor.distance * 100)
    sleep(1)

```
And it gives the error: **ValueError: 'no permission to update GPIO'**

### GPIO Code that works correctly

```
EchoPin = 0
TrigPin = 1

def distance():
    GPIO.output(TrigPin,GPIO.LOW)
    time.sleep(0.000001)  # One micro second
    GPIO.output(TrigPin,GPIO.HIGH)
    time.sleep(0.000001)  # One micro second
    GPIO.output(TrigPin,GPIO.LOW)
    start_waiting_for_echo = time.time()
    while not GPIO.input(EchoPin):
        waiting_for_echo = time.time()
        if (waiting_for_echo - start_waiting_for_echo) > 0.03 :
            return -1 # time-out

    echo_start = time.time()
    while GPIO.input(EchoPin):
        waiting_for_echo_end = time.time()
        if(waiting_for_echo_end - echo_start) > 0.03 :
            return -1 # time out

    echo_end = time.time()
    echo_duration = echo_end - echo_start
    time.sleep(0.01)
    return echo_duration * 17150  # Speed of sound * time / 2

```
### Context

I am running a Raspberry Pi 3b+ with ubuntu server. The ultrasonic sensor is indeed connected to pins 0 (Trigger) and 1 (Echo). Elsewhere I have used gpiozero with the PigPioFactory and it has worked well. I am running the service.

**Comment by bennuttall** [comment]
You need to use the lgpio pin factory on Ubuntu.

**Reply by pitosalas** [reply]
Thanks. But PiGPIOFactory is working for the motors, servos, led, beeper, without problem. Just dumb luck?

**Reply by bennuttall** [reply]
Oh I thought it wasn't supported at all. Perhaps it's an older Ubuntu version? @waveform80 what's the state of play?

**Reply by lurch** [reply]
> Thanks. But PigPioFactory is working for the motors, servos, led, beeper, without problem. Just dumb luck?

Careful with your capitalisation! :pig2: :factory:  :joy:

**Reply by pitosalas** [reply]
@lurch @bennuttall @waveform80 Note that I added working code that uses RPi.GPIO. I would love to hear some tip of what I am doing wrong. I tried changing the pin factory to lgpio and I got the same error. And of course as I mentioned, using PiGPIOFactory for led, servo, motor, beeper all work fine.
